tests/python_functional/src/helpers/loggen/conftest.py

```diff
@@ -0,0 +1,160 @@
+            start_parameters.append("--stream")
+
+        if udp is True:
+            start_parameters.append("--dgram")
+
+        if use_ssl is True:
+            start_parameters.append("--use-ssl")
+
+        if dont_parse is True:
+            start_parameters.append("--dont-parse")
+
+        if read_file is True:
+            start_parameters.append("--read-file={}".format(read_file))
+
+        if skip_tokens is not None:
+            start_parameters.append("--skip-tokens={}".format(skip_tokens))
+
+        if loop_reading is True:
+            start_parameters.append("--loop-reading")
+
+        if rate is not None:
+            start_parameters.append("--rate={}".format(rate))
+
+        if interval is not None:
+            start_parameters.append("--interval={}".format(interval))
+
+        if permanent is True:
+            start_parameters.append("--permanent")
+
+        if syslog_proto is True:
+            start_parameters.append("--syslog-proto")
+
+        if proxied is True:
+            start_parameters.append("--proxied")
+
+        if sdata is True:
+            start_parameters.append("--sdata")
+
+        if no_framing is True:
+            start_parameters.append("--no-framing")
+
+        if active_connections is not None:
+            start_parameters.append("--active-connections={}".format(active_connections))
+
+        if idle_connections is not None:
+            start_parameters.append("--idle-connections={}".format(idle_connections))
+
+        if ipv6 is True:
+            start_parameters.append("--ipv6")
+
+        if debug is True:
+            start_parameters.append("--debug")
+
+        if number is not None:
+            start_parameters.append("--number={}".format(number))
+
+        if csv is True:
+            start_parameters.append("--csv")
+
+        if quiet is True:
+            start_parameters.append("--quiet")
+
+        return start_parameters
+
+    def start(
+        self, target, port, inet=None, unix=None, tcp=None, udp=None, use_ssl=None, dont_parse=None, read_file=None, skip_tokens=None, loop_reading=None,
+        rate=None, interval=None, permanent=None, syslog_proto=None, proxied=None, sdata=None, no_framing=None, active_connections=None,
+        idle_connections=None, ipv6=None, debug=None, number=None, csv=None, quiet=None,
+    ):
+
+        if self.loggen_proc is not None:
+            return
+
+        self.parameters = self._decode_start_parameters(
+            inet, unix, tcp, udp, use_ssl, dont_parse, read_file, skip_tokens, loop_reading,
+            rate, interval, permanent, syslog_proto, proxied, sdata, no_framing, active_connections,
+            idle_connections, ipv6, debug, number, csv, quiet,
+        )
+
+        self.loggen_proc = ProcessExecutor().start(
+            [self.loggen_bin_path] + self.parameters + [target, port],
+            self.loggen_stdout_path,
+            self.loggen_stderr_path,
+        )
+
+        return self.loggen_proc
```

Comment:
What happens, when the `loggen` run is finished by its own? Can we run it again? I think we cannot, because `self.loggen_proc` is not set to None.

Reply:
with the latest commits it is OK. you can call the start again and it will start another instance of loggen.